packages/common/src/api/account.ts

```diff
@@ -151,73 +155,78 @@ const accountApi = createApi({
+        const encodedManagerUserId = encodeHashId(managerUserId)
         const sdk = await audiusSdk()
 
-        await sdk.grants.approveGrant({
+        await sdk.grants.removeManager({
           userId: encodedUserId,
-          grantorUserId: encodedGrantorUserId
+          managerUserId: encodedManagerUserId
         })
 
         return payload
       },
       options: {
-        idArgKey: 'grantorUser.user_id',
-        type: 'mutation'
+        idArgKey: 'managerUserId',
+        type: 'mutation',
+        schemaKey: 'userManagers'
       },
-      async onQueryStarted(
-        payload: ApproveManagedAccountPayload,
-        { dispatch }
-      ) {
-        const { userId, grantorUser } = payload
+      async onQueryStarted(payload: RemoveManagerPayload, { dispatch }) {
+        const { managerUserId, userId } = payload
         dispatch(
           // TODO(C-4330) - The return typing here for `updateQueryData` is erroneous - fix.
           // @ts-expect-error
           accountApi.util.updateQueryData(
             'getManagedAccounts',
-            { userId },
+            { userId: managerUserId },
             (state) => {
-              const currentTime = dayjs().format('YYYY-MM-DD HH:mm:ss')
               // TODO(C-4330) - The state type is incorrect - fix.
               // @ts-expect-error
-              const foundIndex = state.managedUsers.findIndex(
-                (m: ManagedUserMetadata) =>
-                  m.user.user_id === grantorUser.user_id
+              const foundIndex = state.managedUsers?.findIndex(
+                (m: { user: number }) => m.user === userId
               )
+              if (foundIndex != null && foundIndex > -1) {
+                // @ts-expect-error (C-4330)
+                state.managedUsers.splice(foundIndex, 1)
+              }
+            }
+          )
+        )
+        dispatch(
+          // @ts-expect-error
+          accountApi.util.updateQueryData(
+            'getManagers',
+            { userId },
+            (state) => {
+              // TODO(C-4330) - The state type is incorrect - fix.
               // @ts-expect-error
-              state.managedUsers.splice(foundIndex, 1, {
-                grant: {
-                  created_at: currentTime,
-                  // TODO(nkang - C-4332) - Fill this in
-                  grantee_address: '',
-                  is_approved: true,
-                  is_revoked: false,
-                  updated_at: currentTime,
-                  user_id: userId
-                },
-                user: grantorUser
-              })
+              const foundIndex = state.userManagers?.findIndex(
+                (m: { manager: number }) => {
+                  return m.manager === managerUserId
+                }
+              )
+              if (foundIndex != null && foundIndex > -1) {
+                // @ts-expect-error (C-4330)
+                state.userManagers.splice(foundIndex, 1)
```

Comment:
This pattern feels cumbersome to have to implement for every endpoint that modifies things. I'm wondering if we can come up with an easier way to do this, since we need optimistic updates for lots of things.

Reply:
yeah @amendelsohn and i briefly discussed this yesterday! might be something we consider for Audius-query v2